pkg/network/kubemacpool.go

```diff
@@ -84,6 +71,28 @@ func renderKubeMacPool(conf *opv1alpha1.NetworkAddonsConfigSpec, manifestDir str
 	return objs, nil
 }
 
+func fillDefaultsKubeMacPool(conf, previous *opv1alpha1.NetworkAddonsConfigSpec) error {
+	if conf.KubeMacPool.RangeStart == "" || conf.KubeMacPool.RangeEnd == "" {
+		if previous != nil && previous.KubeMacPool != nil {
+			conf.KubeMacPool = previous.KubeMacPool
+			return nil
+		}
+
+		prefix, err := generateRandomMacPrefix()
```

Comment:
please add an inline comment that if no range was specified, we generated a random prefix